### PR TITLE
Look for the nddscpp CMake module

### PR DESCRIPTION
**Pull request by esteve**

### CODE REVIEW

**Comment by wjwwood** [comment]
Testing here: http://54.183.26.131:8080/job/ros2_batch_ci_test_linux/28/console

**Comment by wjwwood** [comment]
Now we get a meaningful error w.r.t. pkg-config:

```
CMake Error at /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:106 (find_package_handle_standard_args)
  /usr/share/nddscpp/cmake/nddscppConfig.cmake:8 (find_package)
  cmake/Modules/FindConnext.cmake:220 (find_package)
  CMakeLists.txt:8 (find_package)


-- Configuring incomplete, errors occurred!
```

**Comment by wjwwood** [comment]
I'll run again with pkg-config installed, the deb should be updated to depend on it. I think @esteve may already be away of that since we talked about it this morning.

**Comment by wjwwood** [comment]
New job: http://54.183.26.131:8080/job/ros2_batch_ci_test_linux/29/console

**Comment by wjwwood** [comment]
New problem:

http://54.183.26.131:8080/job/ros2_batch_ci_test_linux/32/console

It doesn't find all of the required libraries. That job is with the four rti binaries (excluding the rti tools deb), NDDS_HOME is not set, pkg-config is explicitly installed, and this pull request's branch is being used.

**Comment by esteve** [comment]
THe ros2_batch_ci_linux job now finds RTI Connext: http://54.183.26.131:8080/view/ros2/job/ros2_batch_ci_linux/15/console

**Comment by esteve** [comment]
@dirk-thomas @tfoote @wjwwood please review

**Comment by wjwwood** [comment]
+1

File: connext_cmake_module/cmake/Modules/FindConnext.cmake [hunk 1]
```diff
@@ -217,14 +216,15 @@ if(NOT "${_NDDSHOME} " STREQUAL " ")
...
-  if(ndds_cpp_FOUND)
-    message(STATUS "Found RTI Connext: ${ndds_cpp_DIR}")
-    set(Connext_INCLUDE_DIRS ${ndds_cpp_INCLUDE_DIRS})
-    set(Connext_LIBRARIES ${ndds_cpp_LIBRARIES})
+  find_package(nddscpp)
+  find_package(rticonnextmsgcpp)
+  if(nddscpp_FOUND AND rticonnextmsgcpp_FOUND)
+    message(STATUS "Found RTI Connext: ${nddscpp_DIR}")
+    set(Connext_INCLUDE_DIRS ${nddscpp_INCLUDE_DIRS})
+    set(Connext_LIBRARIES ${rticonnextmsgcpp_LIBRARIES})
```
Comment (dirk-thomas) [inline connext_cmake_module/cmake/Modules/FindConnext.cmake]: Why does this not contain `nddscpp_LIBRARIES` anymore?

Comment (esteve) [inline connext_cmake_module/cmake/Modules/FindConnext.cmake]: Because `rticonnextmsgcpp_LIBRARIES` already contains them.

Comment (dirk-thomas) [inline connext_cmake_module/cmake/Modules/FindConnext.cmake]: If `rticonnextmsgcpp` contains the superset of `nddscpp` as well as the messaging libraries should we then just `find_package(rticonnextmsgcpp)` and use its variables? The mixture looks weird to me.

Comment (esteve) [inline connext_cmake_module/cmake/Modules/FindConnext.cmake]: We could add `nddscpp_LIBRARIES`, even if it's a bit redundant, and remove them from here when/if the messaging libraries are included in RTI Connext.

Comment (dirk-thomas) [inline connext_cmake_module/cmake/Modules/FindConnext.cmake]: The PR can be merged as-is but this should be cleaned up sometime in the near future (if we don't a Connext version which just solves that in a single package).

**Comment by wjwwood** [comment]
So, are we going to make an adjustment to this or is it good to go? I don't mind either way, just wondering.

**Comment by esteve** [comment]
@wjwwood I'm waiting for @tfoote 's review.

File: connext_cmake_module/cmake/Modules/FindConnext.cmake (hunk 1 above)
Comment (wjwwood) [inline connext_cmake_module/cmake/Modules/FindConnext.cmake]: I agree that's weird that we find package both but use `INCLUDE_DIRS` from one and `LIBRARIES` from another. At the very least maybe @esteve could add a comment that `rticonnextmsgcpp_LIBRARIES` contains the `nddscpp_LIBRARIES` and a `TODO` comment to clean it up?

Comment (esteve) [inline connext_cmake_module/cmake/Modules/FindConnext.cmake]: I ended up adding `nddscpp_LIBRARIES` too, it took less time than adding a comment :-)

**Comment by tfoote** [comment]
+1

**Pull request merged**